core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-variants/dot-experiments-configuration-variants.component.html

```diff
@@ -76,6 +87,38 @@ <h2 class="flex align-items-center gap-1 uppercase">
                 ></button>
             </div>
         </p-card>
+
+        <ng-template #editableVariantNameTpl>
+            <p-inplace closable="closable">
+                <ng-template pTemplate="display">
+                    <span
+                        class="flex flex-grow-1 align-items-center gap-2 variant-name"
+                        data-testId="variant-name"
+                    >
+                        {{ variant.name }} <i class="pi pi-pencil"></i>
+                    </span>
+                </ng-template>
+                <ng-template pTemplate="content">
+                    <input #variantEdited [value]="variant.name" pInputText type="text" />
+                    <button
+                        class="p-button-rounded p-button-text"
+                        #button
+                        [loading]="button.disabled"
+                        (click)="
+                            editVariantName(variantEdited.value, variant); button.disabled = true
```

Comment:
How about doing this in a function? To make it more readable. 

Also, can we explain why we need the `button.disabled = true`? For future reference.